include/NeoFOAM/core/database/database.hpp

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2024 NeoFOAM authors
+
+#pragma once
+
+#include <unordered_map>
+#include <string>
+#include <any>
+#include <memory>
+#include "NeoFOAM/core/database/collection.hpp"
+
+namespace NeoFOAM
+{
+
+
+class Database
+{
+public:
+
+    /**
+     * @brief Inserts a collection into the database with the specified key.
+     *
+     * @param key The key associated with the collection to be inserted.
+     * @param col The collection to be inserted into the database.
+     * @return Collection& A reference to the inserted collection.
+     */
+    Collection& insert(const std::string& key, const Collection& col);
+
+    /**
+     * @brief Checks if the database contains an collection with the specified name.
+     *
+     * @param name The name of the collection to check for.
+     * @return true if the collection exists in the database, false otherwise.
+     */
+    bool contains(const std::string& name) const;
+
+    /**
+     * @brief Removes a collection from the database.
+     *
+     * This function removes the collection with the specified name from the database.
+     *
+     * @param name The name of the collection to be removed.
+     * @return true if the collection was successfully removed, false otherwise.
+     */
+    bool remove(const std::string& name);
+
+    /**
+     * @brief Retrieves a collection by its name.
+     *
+     * This function searches for and returns a reference to a collection
+     * identified by the given name. If the collection does not exist,
+     * the behavior is undefined.
+     *
+     * @param name The name of the collection to retrieve.
+     * @return Collection& A reference to the collection with the specified name.
+     */
+    Collection& getCollection(const std::string& name);
```

Comment:
```suggestion
    Collection& at(const std::string& name);
```
just to be more consistent with the underlying operation.